core/supplydelta/supplydelta.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2022 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package supplydelta
+
+import (
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/consensus/ethash"
+	"github.com/ethereum/go-ethereum/core/state/snapshot"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/ethereum/go-ethereum/trie"
+)
+
+// SupplyDelta calculates the Ether delta across two state tries. That is, the
+// issuance minus the ETH destroyed.
+func SupplyDelta(block *types.Block, parent *types.Header, db *trie.Database, config *params.ChainConfig) (*big.Int, error) {
+	var (
+		supplyDelta = new(big.Int)
+		start    = time.Now()
+	)
+	// Open the two tries.
+	if block.ParentHash() != parent.Hash() {
+		return nil, fmt.Errorf("parent hash mismatch: have %s, want %s", block.ParentHash().Hex(), parent.Hash().Hex())
+	}
+	src, err := trie.New(common.Hash{}, parent.Root, db)
```

Comment:
Not sure about the owner argument here. I copied the empty hash from the changes I saw made in the commit that introduced the new API. No clue if that's correct here. Please review.

Reply:
It's the account-trie, as opposed to a storage-trie belonging to (owned by) an account within the trie. So `common.Hash` is correct here.

Reply:
I don't fully understand the distinction (I can guess at account-trie, but not storage-true), but who knows, perhaps two PRs down the line, this context will help me 😉 .

Thanks for confirming @holiman, the code is happily running in production now, and Geth is configured for the merge 🐼 . Now if only the lighthouse node can sync in time 😬 😄 .